### DISCUSSION TITLE
Boot Performance

### DISCUSSION THREAD

**Discussion by cpxazn** [marcone/teslausb#876] [Q&A]
I'm currently using a Raspberry Pi 4B with my 2018 Model 3, and it takes around 55 seconds before the car recognizes the storage device. I've done some optimization by disabling some services, and when I run `systemd-analyze`, I get `Startup finished in 3.698s (kernel) + 7.417s (userspace) = 11.116s`, so something is definitely dragging. The wiki shows that it can take anywhere from seconds to a minute, but If I can even get down to 20-30 seconds, I'd be pretty happy.

I'd prefer to not always power on the Pi 4 because it produces quite a bit a heat, and keeping sentry mode on 24/7 seems to be quite a waste of power and storage space. 55 seconds can cause me to miss a parking accident, so I'm wondering how long it's taking others to boot up and using what setup?

EDIT: Just adding some more details, I used the prebuilt bullseye image. Using SD as storage is slightly faster than external SSD when booting.

**Comment by marcone** [comment]
You probably want to run something like `systemd-analyze critical-chain teslausb.service` to get more useful info.

Have you tried the Buster image? The Bullseye image isn't very well tested or optimized.
Running the Buster image I see the drives connected to the host about 17 seconds after archiveloop starts (which includes 8 seconds of waiting for autofs to start), which in turn started at 12 seconds uptime.

**Comment by marcone** [comment]
One thing I noticed is that autofs depends on the networking service (presumably because autofs is often used to automount network shares), and the network service can take a while to start up. Modifying the systemd config file for autofs to no longer list that it depends on the network service shaves a few seconds off of TeslaUSB startup time, though some more investigation is needed to see if that's an OK thing to do.
